src/Elliptic/Systems/Xcts/Events/ObserveAdmIntegrals.hpp

```diff
@@ -0,0 +1,194 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <optional>
+#include <pup.h>
+#include <string>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Tags.hpp"
+#include "Domain/Tags/FaceNormal.hpp"
+#include "Domain/Tags/Faces.hpp"
+#include "Elliptic/Systems/Xcts/Tags.hpp"
+#include "IO/Observer/GetSectionObservationKey.hpp"
+#include "IO/Observer/ObservationId.hpp"
+#include "IO/Observer/ObserverComponent.hpp"
+#include "IO/Observer/ReductionActions.hpp"
+#include "IO/Observer/TypeOfObservation.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Options/String.hpp"
+#include "Parallel/ArrayIndex.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Reduction.hpp"
+#include "Parallel/TypeTraits.hpp"
+#include "ParallelAlgorithms/Events/Tags.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Event.hpp"
+#include "Utilities/Serialization/CharmPupable.hpp"
+
+namespace Events {
+
+/// @{
+/*!
+ * \brief Computes the ADM integrals locally (within one element).
+ *
+ * To get the total ADM integrals, the results need to be summed over in a
+ * reduction.
+ *
+ * See `Xcts::adm_linear_momentum_surface_integrand` for details on the formula
+ * for each integrand.
+ */
+void local_adm_integrals(
+    gsl::not_null<tnsr::I<double, 3>*> result,
+    const Scalar<DataVector>& conformal_factor,
+    const tnsr::ii<DataVector, 3>& spatial_metric,
+    const tnsr::II<DataVector, 3>& inv_spatial_metric,
+    const tnsr::ii<DataVector, 3>& extrinsic_curvature,
+    const Scalar<DataVector>& trace_extrinsic_curvature,
+    const InverseJacobian<DataVector, 3, Frame::ElementLogical,
+                          Frame::Inertial>& inv_jacobian,
+    const Mesh<3>& mesh, const Element<3>& element,
+    const DirectionMap<3, tnsr::i<DataVector, 3>>& conformal_face_normals);
+/// @}
+
+/// @{
+/*!
+ * \brief Observe ADM integrals after the XCTS solve.
+ *
+ * Writes reduction quantities:
```

Comment:
Please add that this event takes a surface integral over the outer boundary, which is defined as the domain boundary in upper logical zeta direction.